.github/workflows/release-azure-containerapp-v1.yaml

```
@@ -0,0 +1,376 @@
+            --name "$CONTAINER_APP_NAME" \
+            --query "id" \
+            --output tsv)
+
+          echo "ca_id=$container_app_id" >> $GITHUB_OUTPUT
+
+      - name: Check for New Revision Errors and Increment Traffic
+        if: ${{ steps.get_revision_mode.outputs.revision_mode == 'Multiple' && steps.create_new_revision.outputs.skip_remaining_steps == 'false' }}
+        env:
+          SCRIPT_NAME: ${{ inputs.script_name }}
+          NEW_REVISION_NAME: ${{ steps.create_new_revision.outputs.new_revision_name }}
+          CURRENT_REVISION_NAME: ${{ steps.get_current_revision.outputs.current_revision_name }}
+          FUNCTION_APP_NAME: "io-p-itn-auth-lollipop-func-02"
+          LOG_ANALITYCS_WORKSPACE_ID: "/subscriptions/ec285037-c673-4f58-b594-d7c480da4e8b/resourcegroups/io-p-rg-common/providers/microsoft.operationalinsights/workspaces/io-p-law-common"
+        run: |
+          set -euo
+
+          if [ -z "$SCRIPT_NAME" ]; then
+            echo "::warning::No script provided, skipping canary deployment"
+
+            az containerapp ingress traffic set \
+              --name "$CONTAINER_APP_NAME" \
+              --revision-weight "$CURRENT_REVISION_NAME"=0 \
+              --revision-weight "$NEW_REVISION_NAME"=100
+
+            exit 0
+          fi
+
+          currentPercentage=0
+
+          while true; do
+
+            case "$SCRIPT_LANGUAGE" in
+              "typescript (npm)")
+                output="$(npm run -s "$SCRIPT_NAME" $currentPercentage)"
+                ;;
+              "typescript (yarn)")
+                output="$(yarn "$SCRIPT_NAME" $currentPercentage)"
+                ;;
+              *)
+                echo "::error::script_language input is required"
+                exit 1
+                ;;
+            esac
+
+            exit_code=$?
+            echo "Script Output: $output"
+
+            if [ $exit_code -ne 0 ]; then
+              echo "::error::Monitoring script failed."
+              echo "::warning::Reverting traffic to previous revision."
+
+              az containerapp ingress traffic set \
+                --name "$CONTAINER_APP_NAME" \
+                --revision-weight "$CURRENT_REVISION_NAME"=100 \
+                --revision-weight "$NEW_REVISION_NAME"=0
+              exit 1
+            fi
+
+            # Parse JSON's output using jq.
+            swap=$(echo "$output" | jq -r '.swap // empty')
+            if [ "$swap" == "true" ]; then
+              echo "::warning::Swap to new version requested."
+              echo "::notice::Swapping all traffic to next revision."
+
+              az containerapp ingress traffic set \
+                --name "$CONTAINER_APP_NAME" \
+                --revision-weight "$CURRENT_REVISION_NAME"=0 \
+                --revision-weight "$NEW_REVISION_NAME"=100
+              break
+            fi
+
+            nextIncrementPercentage=$(echo "$output" | jq -r '.nextIncrementPercentage')
```

```suggestion
            nextPercentage=$(echo "$output" | jq -r '.nextIncrementPercentage')
```

if you call it with 'increment' it looks like the value only contains the delta (increment) while if I understand well it represents the whole percentage.

---

Updated